gui/default/syncthing/core/syncthingController.js

```diff
@@ -1732,21 +1732,21 @@ angular.module('syncthing.core')
         };
 
         $scope.setAllDevicesPause = function (pause) {
-            for (var id in $scope.devices) {
-                $scope.devices[id].paused = pause;
-            };
+            $scope.otherDevices().forEach(function (device) {
+                $scope.devices[device.deviceID].paused = pause;
+            });
             $scope.config.devices = deviceList($scope.devices);
             $scope.saveConfig();
         }
 
         $scope.isAtleastOneDevicePausedStateSetTo = function (pause) {
-            for (var id in $scope.devices) {
-                if ($scope.devices[id].paused == pause) {
-                    return true;
+            var paused = false;
+            $scope.otherDevices().forEach(function (device) {
+                if ($scope.devices[device.deviceID].paused == pause) {
+                    paused = true;
                 }
-            }
-
-            return false
+            });
+            return paused;
```

Comment:
I'd just `continue` if `== $scope.myID` for less iteration.

Reply:
To be fair, I'm not really sure where I should add the check. Do you mean that it should be added to the current `forEach` method, or rather that the current one should be replaced with a different code?

Reply:
Replace the foreach, basically going back to the previous code, with added handling `myID`. Something like the following (beware of mistakes - I wrote it in the github text editor).

```js
            for (var id in $scope.devices) {
                if ($scope.devices[id].paused === pause && id !== $scope.myID) {
                    return true;
                }
            }
            return false
```

It was a early return in a single loop before. Now it's one full loop in `otherDevices`, then another full loop over it's results, even if `paused` is already true.